test/suites/integration/bsp/debt-collection.test.ts

```diff
@@ -181,27 +152,533 @@ describeBspNet(
         paymentStreamInfo.unwrap().priceIndexWhenLastCharged.lt(lastChargeableInfo.priceIndex)
       );
 
-      // Check that Alice now owes the provider.
+      // Check that the user now owes the provider.
       usersWithDebtResult = await userApi.call.paymentStreamsApi.getUsersWithDebtOverThreshold(
         ShConsts.DUMMY_BSP_ID,
         1
       );
       assert(usersWithDebtResult.isOk);
       assert(usersWithDebtResult.asOk.length === 1);
-      assert(usersWithDebtResult.asOk[0].toString() === alice);
+      assert(usersWithDebtResult.asOk[0].toString() === userAddress);
 
+      // Check that the three Providers have tried to charge the user
+      // since the user has a payment stream with each of them
       await userApi.assert.extrinsicPresent({
         method: "chargePaymentStreams",
         module: "paymentStreams",
         checkTxPool: true,
-        assertLength: 1
+        assertLength: 3
       });
 
-      // Seal a block to allow the BSP to charge the payment stream
+      // Seal a block to allow BSPs to charge the payment stream
       await userApi.sealBlock();
 
       // Assert that event for the BSP charging its payment stream was emitted
       await userApi.assert.eventPresent("paymentStreams", "PaymentStreamCharged");
     });
+
+    it("BSP correctly deletes all files from an insolvent user", async () => {
+      ///////** INITIAL SETUP **///////
+
+      // Make sure the payment stream between Alice and the DUMMY_BSP_ID actually exists
+      const paymentStreamExistsResult =
+        await userApi.call.paymentStreamsApi.getUsersOfPaymentStreamsOfProvider(
+          ShConsts.DUMMY_BSP_ID
+        );
+      // Check if the first element of the returned vector is the user
+      assert(paymentStreamExistsResult[0].toString() === userAddress);
+      assert(paymentStreamExistsResult.length === 1);
```

Comment:
If you have to do this as initial steps of every test, you can use the `beforeEach` hook.

Reply:
I don't HAVE to do it as initial steps since checking once in the first test is enough, but I did it here just to be safe. I'm reluctant to add it as a beforeEach hook just because a future test might need for the payment stream to NOT exist